lib/helpers.coffee

```diff
@@ -11,17 +16,88 @@ Helpers = module.exports =
       result.class = result.type.toLowerCase().replace(' ', '-')
       Helpers.validateResults(result.trace) if result.trace
     results
+
+  # Validates that a provider is capable of providing the Base Linter info
+  #   needed to be used at lint time.
   validateLinter: (linter) ->
     unless linter.grammarScopes instanceof Array
-      message = "grammarScopes is not an Array. (see console for more info)"
-      console.warn(message)
-      console.warn('grammarScopes', linter.grammarScopes)
-      throw new Error(message)
-
+      throw new Error("grammarScopes is not an Array. Got: #{linter.grammarScopes})")
     unless linter.lint?
       throw new Error("Missing linter.lint")
-
     if typeof linter.lint isnt 'function'
       throw new Error("linter.lint isn't a function")
-
     return true
+
+  # Everything past this point relates to CLI helpers as loosly demoed out in:
+  #   https://gist.github.com/steelbrain/43d9c38208bf9f2964ab
+
+  exec: (command, options = {stream: 'stdout'}) ->
+    throw new Error "Nothing to execute." if not arguments.length
+    return new Promise (resolve, reject) ->
+      process = child_process.exec(command, options)
```

Comment:
Don't use `exec`, use [`spawn`](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options) instead. With `exec` it forces you to combine command and arguments into one big string that is then going to get split and you have to use special handling to make spaces work correctly. `spawn` takes a command and an array of arguments.